app/queries/servers/thread.ts

```diff
@@ -101,6 +101,8 @@ export const queryThreadsInTeam = ({database, hasReplies = true, isFollowing = t
 
     if (onlyUnreads) {
         query.push(Q.where('unread_replies', Q.gt(0)));
+    } else {
+        query.push(Q.where('loaded_in_global_threads', true));
```

Comment:
`else` part might cause an issue when marking the team threads as read. Can we add it conditionally?

Reply:
@anurag6713 I can yes, but why do you think it's going to cause an issue?

Reply:
Yes, going by the code, for now, only one of these two cases is true. If/when the use case comes up, we can make it configurable with a default value.  This is good and doesn't cause an issue 🙂 Issue was in my code and I fixed it (marking threads received through WS as `loaded_in_global_thread`: true)

Reply:
We'll need to add those threads conditionally though.
If a thread received is newer than any of the threads we already have in with `loaded_in_global_threads: true` then it's safe to add. BUT if a thread is older we can't add it, since it can break pagination if it's added at the end of the list.

Reply:
Agreed! Can we include those checks as part of this PR? I will sync my other changes from open PR to our working draft PR `MM-39707_gekidou_crt` and keep it up to date.

Reply:
I'm expecting to see changes here in another PR then

Reply:
@enahum the change should be on the WS event when a new thread is received.

Adding some info for clarity:

The convention is that only threads being in the `ThreadsInTeam` will be shown in a thread list,
and that only threads marked as `loaded_in_global_threads` are being shown in the `All threads` tab in the list.

So when a thread arrives through different means, whether it's a WS event, or just fetching a (*new) thread by opening it in a channel, etc... we'll need to check if it's newer than any of the existing threads in the `all threads` list.
If it is it will be added in the `ThreadsInTeam` and will be marked as `loaded_in_global_threads: true`.
If it's not newer but it is an unread thread it will still be added in the `ThreadsInTeam` but marked as `loaded_in_global_threads: false`.
Does it make sense?

I will add a new task about this. 

*not existing in the DB yet

PS: all this is done to avoid creating gaps in pagination in the `all threads` tab.

Reply:
Having said that I realized that we need to move `loaded_in_global_threads` from the `Thread` to the `ThreadsInTeam` table.
Should I open a new PR?

Reply:
> Having said that I realized that we need to move `loaded_in_global_threads` from the `Thread` to the `ThreadsInTeam` table.

Opened a PR here https://github.com/mattermost/mattermost-mobile/pull/6100